src/EventListener/AbstractSubscriber.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusFacebookPlugin\EventListener;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Setono\SyliusFacebookPlugin\Context\PixelContextInterface;
+use Setono\SyliusFacebookPlugin\DataMapper\DataMapperInterface;
+use Setono\SyliusFacebookPlugin\Factory\PixelEventFactoryInterface;
+use Setono\SyliusFacebookPlugin\ServerSide\ServerSideEventFactoryInterface;
+use Symfony\Bundle\SecurityBundle\Security\FirewallMap;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+abstract class AbstractSubscriber implements EventSubscriberInterface
+{
+    protected PixelContextInterface $pixelContext;
+
+    protected RequestStack $requestStack;
+
+    protected FirewallMap $firewallMap;
+
+    protected ServerSideEventFactoryInterface $serverSideFactory;
+
+    protected DataMapperInterface $dataMapper;
+
+    protected PixelEventFactoryInterface $pixelEventFactory;
+
+    protected EntityManagerInterface $entityManager;
+
+    public function __construct(
+        PixelContextInterface $pixelContext,
+        RequestStack $requestStack,
+        FirewallMap $firewallMap,
+        ServerSideEventFactoryInterface $serverSideFactory,
+        DataMapperInterface $dataMapper,
+        PixelEventFactoryInterface $pixelEventFactory,
+        EntityManagerInterface $entityManager
+    ) {
+        $this->pixelContext = $pixelContext;
+        $this->requestStack = $requestStack;
+        $this->firewallMap = $firewallMap;
+        $this->serverSideFactory = $serverSideFactory;
+        $this->dataMapper = $dataMapper;
+        $this->pixelEventFactory = $pixelEventFactory;
+        $this->entityManager = $entityManager;
+    }
+
+    protected function getMasterRequest(): ?Request
+    {
+        return $this->requestStack->getMasterRequest();
+    }
+
+    protected function isShopContext(Request $request = null): bool
+    {
+        if (null === $request) {
+            $request = $this->requestStack->getCurrentRequest();
+            if (null === $request) {
+                return true;
+            }
+        }
+
+        $firewallConfig = $this->firewallMap->getFirewallConfig($request);
+        if (null === $firewallConfig) {
+            return true;
+        }
+
+        return $firewallConfig->getName() === 'shop';
+    }
+
+    /**
+     * @param object $source
```

Comment:
What's the reason for not type hinting with `object` directly in the method signature?

Reply:
Already fixed

Reply:
Not fixed?